lib/runner.js

```diff
@@ -378,7 +379,15 @@ Runner.prototype.hook = function(name, fn) {
       }
       if (err) {
         if (err instanceof Pending) {
-          if (name === HOOK_TYPE_BEFORE_EACH || name === HOOK_TYPE_AFTER_EACH) {
+          if (name === HOOK_TYPE_AFTER_EACH || name === HOOK_TYPE_AFTER_ALL) {
+            utils.deprecate(
+              format(
+                "skipping a test within an '%s' hook is DEPRECATED and will be removed from a future version of Mocha. " +
+                  "Use 'return' instead to exit hook abortively.",
+                name
+              )
+            );
+          } else if (name === HOOK_TYPE_BEFORE_EACH) {
```

Comment:
I thought the idea with the deprecation would be to just _warn_ the user and keep the logic otherwise the same.  as I said, I'm not sure what this would necessarily break.  unless we can show that there's a reproducible bug caused by the current behavior, I think I'd prefer we just warn...

Reply:
On the flip side, this could be viewed as a fix to an existing **bug**, no deprecation message -- just throw, and document the condition as a breaking change. This _is_ being put into the forthcoming major release after all.

Reply:
Out of curiousity, what _is_ the LOE of another 5.x release? Could several of these deprecations be trivially backported and released? Then we could apply the changes to 6.0 as they should be without having to drag deprecated behavior around for yet another year...

Reply:
With a deprecation message only and no changes to the logic, we are on the safe side. Users get opportunity to give their feedback, just in case (I don't expect any). This issue is an old one, so there is no need to push quick changes.